src/test/regress/Makefile

```diff
@@ -212,6 +212,9 @@ check-citus-upgrade-mixed-local:
 		--citus-old-version=$(citus-old-version) \
 		--mixed
 
+clean-upgrade-artifacts:
+	rm -rf $(citus_abs_srcdir)/tmp_citus_tarballs/ $(citus_abs_srcdir)/tmp_citus_upgrade/ /tmp/citus_copy/
```

Comment:
One downside of removing everything is that we need to download the old versions again and again. The current version doesn't download the old version if it already exists. 

https://github.com/citusdata/citus/blob/1253eeb9ff37206d276910c87e6fefa44d28eec6/src/test/regress/upgrade/generate_citus_tarballs.sh#L52-L54

As the problem is that citus_master folder will not be empty, we could do the following to solve the problem I think:

```bash
  cd "$citus_repo" && rsync -a --exclude='.git' . /tmp/citus_copy
  rm -rf "${basedir}/citus_${citus_version}"
```
https://github.com/citusdata/citus/blob/master/src/test/regress/upgrade/generate_citus_tarballs.sh#L34 (by replacing this line with above)

What do you think?

Reply:
@hanefi ping

Reply:
one problem that I see now is that if we compile with different postgres versions for a specific citus version (say 9.0.0) then we might skip tar generation even though the existing one is compiled with a different pg major.

So your solution might be a better way.

Reply:
>So your solution might be a better way.

I am happy to hear this. I can rebase and merge if you think it makes sense.

Another alternative can be adding PG majors to tarball names. That way, we can safely cache the tarballs I guess.

FYI, I no longer use a Mac, so I can not test if this will work on that environment now.

Reply:
You will also need to add this artifact to `mixed` mode target.

Also can you remove this check:

https://github.com/citusdata/citus/blob/1253eeb9ff37206d276910c87e6fefa44d28eec6/src/test/regress/upgrade/generate_citus_tarballs.sh#L52-L54

Reply:
I addressed your feedbacks here. I intentionally did not touch the make targets that are run on the CI. I think that's ok